averbis/core/_rest_client.py

```diff
@@ -1256,6 +1264,13 @@ def fetch_filename(src: Union[Path, IO, str], default_filename: str) -> str:
         if isinstance(source, str) and mime_type is None:
             mime_type = MEDIA_TYPE_TEXT_PLAIN
 
+        if isinstance(source, Cas):
+            if mime_type is not None:
+                raise Exception(
+                    f"Mime-type cannot be set explicitly when the source is a Cas object."
+                )
```

Comment:
If we want to upload a CAS object, why don't we support setting the mime_type to `application/vnd.uima.cas+xmi`?

Reply:
We don't need the caller to do it, we do it internally. We could allow that the user explicitly sets the value to `application/vnd.uima.cas+xmi` but not to any other value (because other CAS formats are not supported on the Python side) - but the user doesn't have to set it to that value because we set it implicitly anyway. Would you prefer if the user can set it only to `application/vnd.uima.cas+xmi` but doesn't have to?

Reply:
Yes. I understand. I think it might be confusing for a user if (s)he sets the correct mime-type for a CAS as `application/vnd.uima.cas+xmi`, but still receives and Exception.

Hence, I would suggest changing that line 1268 to `if mime_type != "application/vnd.uima.cas+xmi":` and to adjust the Exception message. The rest is fine.